src/metatensor/models/utils/compute_loss.py

```diff
@@ -115,6 +116,18 @@ def compute_model_loss(
     # Based on the keys of the targets, get the outputs of the model:
     model_outputs = _get_model_outputs(model, systems, list(targets.keys()))
 
+    # modify model predictions and targets where averaging per atom is requested
+    if len(peratom_targets) > 0:
+
+        num_atoms = torch.tensor([len(s) for s in systems]).to(device=device)
+        num_atoms = torch.reshape(num_atoms, (-1, 1))
+
+        for pa_target in peratom_targets:
+            model_outputs[pa_target].block().values /= num_atoms
+            # to prevent averaging builds up over epochs
+            targets = targets.copy()
+            targets[pa_target].block().values /= num_atoms
+
```

Comment:
Please not in-place! Assuming this works, this could generate big issues later when we try to use the same tensors in the same TensorBlock (and we won't know). Can you create a different tensor instead?